components/server/src/bitbucket-server/bitbucket-server-context-parser.ts

```diff
@@ -88,7 +88,11 @@ export class BitbucketServerContextParser extends AbstractContextParser implemen
             return await this.handleNavigatorContext(ctx, user, repoKind, host, owner, repoName, more);
         } catch (e) {
             span.addTags({ contextUrl }).log({ error: e });
-            log.error({ userId: user.id }, "Error parsing Bitbucket context", e);
+            let error = e;
+            if (e.name === "HTTPError") {
+                error = e.status; // we don't want to expose the full error message since it contains credentials
+            }
+            log.error({ userId: user.id }, "Error parsing Bitbucket context", error);
             throw e;
```

Comment:
Will we log the thrown error outside somewhere?

Reply:
How about make it a error class?
```
class BitBucketHttpError extends Error {
  construct(originalErr, message) { ... }
}
function handleBitBucketError(err: Error): Error {
  if (e.name !== "HTTPError") {   return err }
  return new  BitBucketHttpError(e, "Error parsing BB context")
}
```

Reply:
Sounds good. I ended up just making `handleBitBucketError` redact the value we care about instead of introducing a new custom error.